packages/nft-minter/test/globalSetup.ts

```diff
@@ -0,0 +1,79 @@
+/* eslint-disable no-console */
+import { ChildProcess, execSync, spawn } from "child_process";
+import pWaitFor from "p-wait-for";
+import { createPublicClient, defineChain, http } from "viem";
+import { hemi } from "viem/chains";
+
+let childProcess: ChildProcess;
+
+export const setup = async function () {
+  // Kill any other previous fork that could've been running, by checking the port
+  try {
+    console.log("Killing any previous local fork");
+    const pids = execSync("lsof -ti:8545").toString().trim();
+    if (pids) {
+      execSync(`echo "${pids}" | xargs kill`);
+    }
+  } catch {
+    // eslint-disable-line no-empty
+  }
+  console.log("Locating anvil binary...");
+  // Locate the anvil binary dynamically, searching in $HOME/.foundry and $HOME/.config/.foundry
+  // if not found there, assume it is not installed. This is because on Github Actions, the installation script
+  // adds anvil into a PATH that is a different shell from the one running here. But as the binary is installed
+  // we just need to find it and execute it. This snippet below does that
+  const anvilPath = execSync(
+    "find $HOME/.foundry $HOME/.config/.foundry -type f -name anvil -perm -u=r 2>/dev/null | head -n 5",
+  )
+    .toString()
+    .split("\n")
+    .filter(Boolean)[0];
+  if (!anvilPath) {
+    // Should always be found - it should be installed with the pretest command
+    throw new Error("Anvil binary not found in $HOME");
+  }
+  console.log(`Found anvil binary at ${anvilPath}`);
+  console.log("Starting local fork on port 8545");
+
+  // Spawn the local fork in the background
+  childProcess = spawn(anvilPath, ["--fork-url", hemi.rpcUrls.default.http[0]]);
+
+  console.log(`Started local fork with pid ${childProcess.pid}`);
```

Comment:
What about calling `npm run fork:stop`, `npm run fork:start` here, to reuse the logic instead of repeating it here, in the NPN scripts and the install script?

It looks like the proper "pre-test" script should 1) check if anvil is installed or do the install, 2) kill any fork, 3) start a new fork.

Reply:
You're right. I guess it took this structure as the code was evolving locally (to work), and then with the addition of the script (to make it work in GH actions), it ended up split up in pieces. I moved the installation, start, and stop into bash scripts. I left the pinging in the setup only.

Changes in [f51c5db](https://github.com/hemilabs/tge-mint-page/pull/2/commits/f51c5db88a6d0011f57a6d1485de0eb456567d60)